src/server/game/Entities/Player/Player.cpp

```diff
@@ -16440,12 +16441,16 @@ bool Player::HasQuestForItem(uint32 itemid) const
 
         QuestStatusData const& q_status = qs_itr->second;
 
-        if (q_status.Status == QUEST_STATUS_INCOMPLETE)
+        if (q_status.Status == QUEST_STATUS_INCOMPLETE || q_status.Status == QUEST_STATUS_COMPLETE)
         {
             Quest const* qinfo = sObjectMgr->GetQuestTemplate(questid);
             if (!qinfo)
                 continue;
 
+            if (questid == excludeQuestId)
+                excludeQuestId = 0;
+                return true;
```

Comment:
Please debug your code line by line and add breakpoints to ensure all cases are tested.
There's no need to say bad words to who is reviewing your code for mistakes you made.